client/me/purchases/manage-purchase/purchase-meta.jsx

```diff
@@ -284,52 +283,53 @@ function PurchaseMetaPrice( { purchase } ) {
 		return translate( 'Free with Plan' );
 	}
 
-	if ( purchase.billPeriodLabel ) {
-		switch ( purchase.billPeriodLabel ) {
-			case 'per year':
-				period = translate( 'year' );
-				break;
-			case 'per month':
-				period = translate( 'month' );
-				break;
-			case 'per week':
-				period = translate( 'week' );
-				break;
-			case 'per day':
-				period = translate( 'day' );
-				break;
+	const getPeriod = () => {
+		if ( purchase.billPeriodLabel ) {
+			switch ( purchase.billPeriodLabel ) {
+				case 'per year':
+					return translate( 'year' );
+				case 'per month':
+					return translate( 'month' );
+				case 'per week':
+					return translate( 'week' );
+				case 'per day':
+					return translate( 'day' );
+			}
 		}
-	}
 
-	if ( plan && plan.term ) {
-		switch ( plan.term ) {
-			case TERM_BIENNIALLY:
-				period = translate( 'two years' );
-				break;
+		if ( plan && plan.term ) {
+			switch ( plan.term ) {
+				case TERM_BIENNIALLY:
+					return translate( 'two years' );
+				case TERM_MONTHLY:
+					return translate( 'month' );
+			}
+		}
 
-			case TERM_MONTHLY:
-				period = translate( 'month' );
-				break;
+		if ( isEmailMonthly( purchase ) ) {
+			return translate( 'month' );
 		}
-	}
 
-	if ( isEmailMonthly( purchase ) ) {
-		period = translate( 'month' );
-	}
+		throw new Error( 'An unexpected error occured' );
```

Comment:
Even though we hope that this never happens, it might be a good idea to give more context in this error message so that it's easier to track down if we ever see it in the wild.

Maybe something like, "Failed to get a billing term for '${productSlug}'".